### ISSUE TITLE
As Open Data Hub Manager I want that georeferenced data are rendered also on a map, to understand how data are distributed from a geographical point of view.

### ISSUE THREAD

**Issue by sseppi** [noi-techpark/it.bz.opendatahub.databrowser#277]
This issue includes:
- [x] rendering on a map at a Dataset Overview level;
- [x] rendering on a map the record displayed in the Table View.

**Comment by pkritzinger** [comment]
@sseppi @gappc 
here's the link to the design of the map-view. 
https://www.figma.com/file/DmuP6Dbv5LzkCNOXrDnWIL/2023?type=design&node-id=1332%3A284&mode=design&t=1S5w2e5Rzyge38pc-1

feel free to get back to us regarding it after having discussed performance-topics with @RudiThoeni

**Comment by mrabans** [comment]
@sseppi we need to evaluate this with @RudiThoeni

**Comment by sseppi** [comment]
As discussed in the yesterday's meeting in this iteration won't be included the following features that will be implemented in the next iterations.

- the filters appliend in the dataset overview will be kept once the user opens the map overview (For this feature a dedicated issue has been created: #607 )
- in the record preview opened on the map more information than the record ID are shown (For this feature a dedicated issue has been created: #60 )

**Comment by gappc** [comment]
@sseppi @mrabans @pkritzinger @Mazzolintis @MatteoBiasi @ohnewein @RudiThoeni: here is the first **pre-alpha** version of the map: https://0.databrowser.gappc.net/?map=true

Please take a look at it and let me know what you think, thx.

**Comment by ohnewein** [comment]
Looks very promising! Even on mobile.

**Comment by ohnewein** [comment]
There is a spelling error in "reference" in the note text.

"Note! Datasets without geographical refer_e_nce will not be shown in map view."

But I would suggest to use a positive frasing and simplify:

"Note! Only datasets containing geographical coordinates can be shown on the map."

**Comment by ohnewein** [comment]
Hi Rudi,

I am not sure, if we should introduce a dedicated performance tuning for specific tools.

It would be better if the Data Browser just uses the standard API as all other developers can. As long as there is not a huge need to improve performance, I would suggest to use standard API calls.

Best regards 
Patrick

**Comment by gappc** [comment]
@ohnewein @sseppi @mrabans @RudiThoeni @MatteoBiasi @pkritzinger @Mazzolintis I've improved the map view implementation, it provides the following features:

- show map with filter
- display datasets on map
- display record / cluster detail (click on marker on the map) as requested in #608 
- linkable URL contains map center + zoom, selected datasets and selected record
- mobile support

You can find a deployment of the current version here: https://0.databrowser.gappc.net/?map=true

The associated PR is here: #681 

Please give it a try and let me know what you think of it.

**Comment by ohnewein** [comment]
I would like to find a way which does not need human input in managing and updating meta data. 

My question is, in scenario (1), how does the map view recognize there is no gps information in the dataset? Could this be also used to recognize if the dataset should be included in the dataset selection list?

**Comment by gappc** [comment]
@ohnewein thx for your feedback 👍 

In general I agree that it would be best to find a way which does not need human input to manage datasets with GPS coordinates.

In solution 1. above, the user would just select any dataset and if that dataset contains GPS info, its records are shown on the map. If there is no GPS info, no datasets are shown on the map. Although that's the easiest solution because we have nothing to change, its also the worst solution in terms of UX.

Given that we want to have proper UX and that we don't want to configure manually which datasets have GPS information, I think we should aim to implement solution 2., what do you say?

**Comment by gappc** [comment]
@RudiThoeni @sseppi @mrabans @ohnewein @MatteoBiasi @pkritzinger @Mazzolintis: the map is now online in the testing environment: https://databrowser.opendatahub.testingmachine.eu

**Comment by ohnewein** [comment]
> 
> Given that we want to have proper UX and that we don't want to configure manually which datasets have GPS information, I think we should aim to implement solution 2., what do you say?


What about a combination? Let's introduce a No-GPS-Data tag. In this way we could exclude those datasets which do NOT have gps data from the selection list. This could reduce the human work and make sure there are no datasets missing from the UI just because no human remembered to tag the dataset correctly.

Would that be a good approach?

**Comment by gappc** [comment]
@ohnewein if I understand you correctly, the idea is that we assume all datasets have GPS information and then manually label those that don’t.

Advantage: by default, a dataset shows up in the map view, unless labeled as not having GPS-Data
Disadvantage: manual labeling still needed for datasets that should not be displayed in map view 

Is that correct?

**Comment by ohnewein** [comment]
Yes. Exactly.

Of course if there exists an option in which no manual labelling is needed, that would be even better.

**Comment by sseppi** [comment]
> ok so we store this Info into the Metadata Api ? a simple Boolen Field `NoGps` other options would be to add an Array which contains Tags

@RudiThoeni and @gappc If we do so, would it be possible, to implement an elaboration that runs once a day sets automatically the ``ǸoGPS``` to true if one of the following condition is verified:

- the JSON schema doesn't foresee any GPS attribute;
- the GPS attribute is null for all records included in the dataset.

Since I think this is a feature that isn't included in this issue. My sugestion would be to, go online with this version of the map and open a new issue to finde the right solution and implement the representation of datasets without GPS.

Do you agree?

**Comment by RudiThoeni** [comment]
yes a scheduled job can always be added, but i would say let's start simple and 

- add this field/tag to the metadata api
- filter the map datasets on this tag

next can be this scheduled job..... this runs completely on the backend so in the databrowser no work has to be done

In which releases we want to deploy this funcitonalities is up to you @sseppi  @gappc .......

**Comment by sseppi** [comment]
> yes a scheduled job can always be added, but i would say let's start simple and
> 
> * add this field/tag to the metadata api
> * filter the map datasets on this tag
> 
> next can be this scheduled job..... this runs completely on the backend so in the databrowser no work has to be done
> 
> In which releases we want to deploy this funcitonalities is up to you [@sseppi](https://github.com/sseppi) [@gappc](https://github.com/gappc) .......

@gappc I would discuss this issue in the next refinement and plan the implementation in the next release.

**Comment by gappc** [comment]
@RudiThoeni @sseppi what's the state of this issue?

**Comment by gappc** [comment]
@sseppi yes, I agree, I think we should put new features into other issues, because this issue here is already quite long and the requested features are implemented and merged.

**Comment by sseppi** [comment]
@gappc OK, I will create the new issue and close this one
